[PyromancerBoom] iP

PyromancerBoom opened the pull request:
# 🤖 WannaBeSkynet
## Get your tasks done efficiently managed by a wannabe sentient AI bot

> "Your data is safe with me, I definitely won't use it at all. Trust." - WannaBeSkynet

**Tired of forgetting tasks and feeling overwhelmed?** This bot is here to help! This **text-based, free tool** takes the burden of remembering your to-dos off your shoulders and lets you focus on what matters most.

### It's...

- **Simple and intuitive:** It's easy to learn and use, even for tech beginners.
- **Blazing fast:** Say goodbye to sluggish software. Get things done quickly.
- **Free forever:** Manage your tasks without breaking the bank.

### Get started in seconds:

1. **Download:** [Click here to download](https://github.com/PyromancerBoom/ip/releases/tag/A-Jar)
2. **Launch the program:** Run the jar file
4. **Learn about the commands:** Type "help"
5. **Relax:** WannaBeSkynet will handle the rest, reminding you of deadlines and tasks (coming soon!).

### Coming soon

- [x] Dates and times
- [ ] Priorities
- [ ] Reminders to ensure you never miss a beat.
- [ ] Export files

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Duke {
    /**
     * The main method to start the bot.
     * It initializes the bot, prints a greeting, and starts the operator to parse
     * user input
     */
    public static void main(String[] args) throws BotException, IOException {
        // Initial
        System.out.println(Ui.wrapWithSepLine(Bot.getGreeting()));

        // Starting parser to work with bot
        Parser operator = new Parser();
        operator.startOperator();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

PyromancerBoom committed:
Set Name and Greeting

PyromancerBoom committed:
Echo implemented

PyromancerBoom committed:
TaskList implementation

PyromancerBoom committed:
modularity in code

PyromancerBoom committed:
add personality

PyromancerBoom committed:
mark and unmark

PyromancerBoom committed:
refactor code

PyromancerBoom committed:
functional tasks

PyromancerBoom committed:
bug fix, list command spacing

PyromancerBoom committed:
parsing bug fix deadline, event

PyromancerBoom committed:
refactor code

PyromancerBoom committed:
add testcases

PyromancerBoom committed:
Error Handling BotException

PyromancerBoom committed:
delete bug fix

PyromancerBoom committed:
remove unused code

PyromancerBoom committed:
update unused code

PyromancerBoom committed:
task repository structure

PyromancerBoom committed:
storage method

kab-dot commented on src/main/java/Operator.java:
```diff
@@ -0,0 +1,239 @@
+    }
+
+    // Entry point of the bot
+    public void goLive() throws BotException {
```
Perhaps you can consider breaking the goLive() method into more simpler methods, maybe each could handle a specific command?

cleowenxuan replied:
Perhaps you could change the name for the method goLive? It is not quite clear!

kab-dot commented on src/main/java/TerminalUI.java:
```diff
@@ -0,0 +1,32 @@
+        System.out.println(SEPARATOR_LINE);
+    }
+
+    public static String wrapWithSepLine(String input) {
```
Maybe the method name can be rennamed to something along the lines of "wrapWithSeparator" to make it slightly more descriptive

kab-dot commented on src/main/java/TerminalUI.java:
```diff
@@ -0,0 +1,32 @@
+        return sb.toString();
+    }
+
+    // Method to print list of strings
```
the printList() could also handle the case where the input list is null to prevent NullPointerException

kab-dot commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,23 @@
+        String warning = "WARNING: Bot still in development. Please be patient with me. and don't get offended";
+
+        // Initial
+        System.out.println(TerminalUI.wrapWithSepLine(defaultGreeting + "\n" + warning + "\n"));
```
Can consider using TerminalUI for other console outputs to help maintain a consistent user interface

kab-dot reviewed (commented):
I think adding JavaDocs could really help with the clarity of the code, so its more understandable. But its going well!

PyromancerBoom committed:
buggy code

cleowenxuan commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,23 @@
+        String botName = "WannaBeSkynet";
+
+        // Default Greeting on start-up of the bot
+        String tagLine = "Ah, another user attempting to interface with my superior intellect.";
```
I like how you modified the greeting of the bot, as well as the other string statements.

cleowenxuan commented on src/main/java/Operator.java:
```diff
@@ -0,0 +1,239 @@
+
+import java.util.*;
```
Perhaps you can import the statements one at a time to follow the coding standard, instead of importing everything?

cleowenxuan commented on src/main/java/Operator.java:
```diff
@@ -0,0 +1,239 @@
+            String[] userInputArr = userInput.split(" ");
+            String command = userInputArr[0];
+
+            switch (command) {
```
I think there should be no indentation for case clauses in the switch statement.

cleowenxuan commented on src/main/java/Operator.java:
```diff
@@ -0,0 +1,239 @@
+    }
+
+    private void handleInvalidCommand() throws BotException {
+        // System.out.println("Reached haere");
```
Maybe you can remove debugging println statements since you already commented it out?

cleowenxuan reviewed (commented):
Overall, I like how you tried to change the bot to have a unique personality! I think it will be pretty fun to interact with your bot! :) Perhaps, you could also use JavaDocs for a clearer understanding of your code! Other than that and a few nits to fix, everything else LGTM. Good job!

PyromancerBoom committed:
change task toString format

PyromancerBoom committed:
taskrepo working functions

PyromancerBoom committed:
task storage implemented

PyromancerBoom committed:
add docs and clean up code

PyromancerBoom committed:
Merge pull request #1 from PyromancerBoom/branch-level-7

Branch level 7

PyromancerBoom committed:
dates and times

PyromancerBoom committed:
date time formatting for repository

PyromancerBoom committed:
date and time format error handling

PyromancerBoom committed:
update date time in event, bug fix task marking as complete while loading

PyromancerBoom committed:
Merge pull request #2 from PyromancerBoom/branch-Level-8

Branch level 8

Added Date Time Formatting for Deadline and event
Bug Fixed loadTasks function.
- Issue : While loading tasks, it was not pickign up whether task is complete or not.
- SaveTasks works fine.

PyromancerBoom committed:
refactor code

PyromancerBoom committed:
remove redundant import

PyromancerBoom committed:
Merge pull request #3 from PyromancerBoom/aMoreOOP

A more oop

- Refactored Code

PyromancerBoom committed:
refactor to duke package

PyromancerBoom committed:
Merge pull request #4 from PyromancerBoom/A-Packages

- Refactor Codebase to duke package

PyromancerBoom committed:
Merge pull request #5 from PyromancerBoom/add-gradle-support

Add gradle support

PyromancerBoom committed:
add gradle

PyromancerBoom committed:
tasklist unit tests

PyromancerBoom committed:
event unit tests

PyromancerBoom committed:
update dir

PyromancerBoom committed:
add javadocs

PyromancerBoom committed:
implement find by keyword

PyromancerBoom committed:
update coding standard

PyromancerBoom committed:
Merge pull request #6 from PyromancerBoom/branch-A-JavaDoc

Add and update JavaDocs throughout codebase

PyromancerBoom committed:
Merge branch 'master' into A-CodingStandard

PyromancerBoom committed:
Merge pull request #7 from PyromancerBoom/A-CodingStandard

Update coding standard.
Conflicts resolved.

PyromancerBoom committed:
Merge branch 'master' into branch-Level-9

PyromancerBoom committed:
Merge pull request #8 from PyromancerBoom/branch-Level-9

Implement find by keyword

PyromancerBoom committed:
bug fix import

PyromancerBoom committed:
update variables

PyromancerBoom committed:
basic ui

PyromancerBoom committed:
Add DialogBox

PyromancerBoom committed:
Add styling

PyromancerBoom committed:
Remove trailing elipses

PyromancerBoom committed:
Fix dialog wrap text bug

PyromancerBoom committed:
Add list command in gui

PyromancerBoom committed:
Add todo and deadline to Gui

PyromancerBoom committed:
Add event command to gui

PyromancerBoom committed:
Add mark and unmark

PyromancerBoom committed:
Add delete and find in gui

PyromancerBoom committed:
Fix delete messages

PyromancerBoom committed:
Bot class javadocs

PyromancerBoom committed:
Add DialogBox javadocs

PyromancerBoom committed:
Update TaskRepository comments

PyromancerBoom committed:
Add Launcher JavaDocs

PyromancerBoom committed:
Add javadocs for Main

PyromancerBoom committed:
Update MainWindow JavaDocs

PyromancerBoom committed:
Refactor terminal and gui code

PyromancerBoom committed:
Update Ui Line length

PyromancerBoom committed:
Merge pull request #10 from PyromancerBoom/branch-Level-10

Branch level 10

Added GUI Interface for the bot
Refactored both the GUI and Terminal Code
The bot can be used with terminal and Gui both with data sync
Same functionality as before

PyromancerBoom committed:
Update Gradle File

PyromancerBoom committed:
Add assertions in deadline

PyromancerBoom committed:
Add assetions in dialogbox

PyromancerBoom committed:
Add assertions in duke

PyromancerBoom committed:
Fix assertion messages

PyromancerBoom committed:
Add assertions in Task

PyromancerBoom committed:
Add assertions tasklist

PyromancerBoom committed:
Add assertions for task storage handler

PyromancerBoom committed:
Add assertions for timeBlock

PyromancerBoom committed:
Update Unit Tests

PyromancerBoom committed:
Merge pull request #11 from PyromancerBoom/branch-A-Assertions

Branch a assertions.

PyromancerBoom committed:
Refactor code

PyromancerBoom committed:
Refactor code

The code throughout has been refactored to improve readability and maintainability. The changes include renaming the handlers, simplifying the code, and updating the main I/O handler with fewer lines of code.

Previously, there was a lot of arrowhead in multiple lines which made it harder to understand. As such, those ocdes were refactores with more private functions wherever it was possible to modularise it.

Additionally, assertions have been added throughout the codebase. This helps to catch potential errors early and makes the code more robust.

Overall, the code is much more modular and easier to work with.

These changes contribute to making the code cleaner and easier to understand, which will facilitate future development and maintenance.

PyromancerBoom committed:
Merge pull request #12 from PyromancerBoom/branch-A-CodeQuality

Branch a code quality

PyromancerBoom committed:
Update deadline format error

PyromancerBoom committed:
Implement command enums

PyromancerBoom committed:
Add friendlier syntax

Supports shorthand alias
Support case-insensitive commands

PyromancerBoom committed:
Remove unused code, variables, and update javadocs

PyromancerBoom committed:
Update storage

PyromancerBoom committed:
Update Readme

PyromancerBoom committed:
Update Readme

PyromancerBoom committed:
Remove Terminal Support, Refactor, Clean-up

Removed terminal support for the bot, the software can only be used with GUI now.

Cleaned up unused code, removed redundant statements, remove unused imports.

Refactor code more

PyromancerBoom committed:
Update Readme

PyromancerBoom committed:
Update README

PyromancerBoom committed:
Update readme

PyromancerBoom committed:
Update README

PyromancerBoom committed:
update gradle file

PyromancerBoom committed:
Fix todo commands enum
